### PR TITLE
Add external protobuf dependencies to examples/protocol_buffers

### PR DESCRIPTION
**Pull request by bshimc**
Functionally the changes should be a no-op for the purposes of the tests in this package but they do shake out some of the issues described in (albeit with different symptoms):

https://github.com/bazelbuild/rules_nodejs/issues/988 (Unable to build ts_proto_library with dep...)
https://github.com/bazelbuild/rules_nodejs/issues/1946 (ts_proto_library: Cannot depend on external proto_library)

With potentially related upstream issues:

https://github.com/grpc/grpc-web/issues/584
https://github.com/protocolbuffers/protobuf/issues/6341

Hopefully, this pull request will lower the barrier for folks more familiar with this toolchain (unfortunately that does not describe me) to investigate.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Changes actually work :sad_trombone:

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [X] Other: At this juncture this PR is more of a bug report / feature request that consolidates a repro case for a pair of existing issues.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Other information

**`$ bazel build //:car_proto`**
```
INFO: Invocation ID: 7c7b3259-275c-46da-99f3-627689ffb0aa
INFO: Analyzed target //:car_proto (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:car_proto up-to-date:
  dist/bin/car_proto-descriptor-set.proto.bin
INFO: Elapsed time: 0.201s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
```
**`$ bazel build //:car`**
```
INFO: Invocation ID: 67d6939e-a48b-4b62-99a3-aa9dc6d15f63
INFO: Analyzed target //:car (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:car up-to-date:
  dist/bin/car_pb.js
  dist/bin/car_grpc_web_pb.js
  dist/bin/car_pb.mjs
  dist/bin/car_grpc_web_pb.mjs
  dist/bin/car_pb.d.ts
  dist/bin/car_grpc_web_pb.d.ts
INFO: Elapsed time: 0.178s, Critical Path: 0.01s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
```
**`$ bazel build //... --sandbox_debug`**
```
INFO: Invocation ID: 980df84a-8b88-4123-b136-b902c9ed89e0
INFO: Analyzed 22 targets (0 packages loaded, 0 targets configured).
INFO: Found 22 targets...
ERROR: /Users/bo/src/github.com/bazelbuild/rules_nodejs/examples/protocol_buffers/BUILD.bazel:124:14: Generating Descriptor Set proto_library //:rules_typescript_proto failed (Exit 1): sandbox-exec failed: error executing command
  (cd /private/var/tmp/_bazel_bo/27e51bb136515535a0b392627bd874c9/sandbox/darwin-sandbox/260/execroot/examples_protocol_buffers && \
  exec env - \
    PATH=/bin:/usr/bin:/usr/local/bin \
    TMPDIR=/var/folders/gn/0mnv24n13sb3c8283vn0cncr0000gn/T/ \
  /usr/bin/sandbox-exec -f /private/var/tmp/_bazel_bo/27e51bb136515535a0b392627bd874c9/sandbox/darwin-sandbox/260/sandbox.sb /var/tmp/_bazel_bo/install/3ac6e4bc67346f686d73708a27e81d33/process-wrapper '--timeout=0' '--kill_delay=15' bazel-out/host/bin/external/com_google_protobuf/protoc '--descriptor_set_out=bazel-out/darwin-fastbuild/bin/rules_typescript_proto-descriptor-set.proto.bin' '-Icar.proto=car.proto' '-Itire.proto=tire.proto' --direct_dependencies car.proto:tire.proto '--direct_dependencies_violation_msg=%s is imported, but //:rules_typescript_proto doesn'\''t directly depend on a proto_library that '\''srcs'\'' it.' car.proto tire.proto) sandbox-exec failed: error executing command
  (cd /private/var/tmp/_bazel_bo/27e51bb136515535a0b392627bd874c9/sandbox/darwin-sandbox/260/execroot/examples_protocol_buffers && \
  exec env - \
    PATH=/bin:/usr/bin:/usr/local/bin \
    TMPDIR=/var/folders/gn/0mnv24n13sb3c8283vn0cncr0000gn/T/ \
  /usr/bin/sandbox-exec -f /private/var/tmp/_bazel_bo/27e51bb136515535a0b392627bd874c9/sandbox/darwin-sandbox/260/sandbox.sb /var/tmp/_bazel_bo/install/3ac6e4bc67346f686d73708a27e81d33/process-wrapper '--timeout=0' '--kill_delay=15' bazel-out/host/bin/external/com_google_protobuf/protoc '--descriptor_set_out=bazel-out/darwin-fastbuild/bin/rules_typescript_proto-descriptor-set.proto.bin' '-Icar.proto=car.proto' '-Itire.proto=tire.proto' --direct_dependencies car.proto:tire.proto '--direct_dependencies_violation_msg=%s is imported, but //:rules_typescript_proto doesn'\''t directly depend on a proto_library that '\''srcs'\'' it.' car.proto tire.proto)
google/api/annotations.proto: File not found.
car.proto:3:1: Import "google/api/annotations.proto" was not found or had errors.
INFO: Elapsed time: 0.250s, Critical Path: 0.08s
INFO: 6 processes: 6 internal.
FAILED: Build did NOT complete successfully
```

### CODE REVIEW

**Comment by bshimc** [comment]
@googlebot I signed it!

**Comment by alexeagle** [comment]
Could you investigate the failure of `//examples:examples_protocol_buffers` ?
```

ERROR: /tmp/tmp-9611qgyUU7oxzINk/BUILD.bazel:124:1: Generating Descriptor Set proto_library //:rules_typescript_proto failed (Exit 1) protoc failed: error executing command bazel-out/host/bin/external/com_google_protobuf/protoc '--descriptor_set_out=bazel-out/k8-fastbuild/bin/rules_typescript_proto-descriptor-set.proto.bin' '-Icar.proto=car.proto' '-Itire.proto=tire.proto' ... (remaining 5 argument(s) skipped)
--
  |  
  | Use --sandbox_debug to see verbose messages from the sandbox
  | google/api/annotations.proto: File not found.
  | car.proto:3:1: Import "google/api/annotations.proto" was not found or had errors.

```

**Comment by alexeagle** [comment]
oh @bshimc I think you're asking for someone on our team to diagnose the failure? maybe @mrmeku

**Comment by bshimc** [comment]
@alexeagle that (asking for help) is correct; I'm not familiar enough with the toolchains here to be of much immediate use.

File: examples/protocol_buffers/BUILD.bazel [hunk 1]
```diff
@@ -23,7 +23,10 @@ ts_proto_library(
 proto_library(
     name = "car_proto",
     srcs = ["car.proto"],
-    deps = [":tire_proto"],
+    deps = [
+        ":tire_proto",
+        "@com_google_googleapis//google/api:annotations_proto",
```
Comment (mrmeku) [inline examples/protocol_buffers/BUILD.bazel:28]: You added the dependency to the wrong proto_library
See https://github.com/bazelbuild/rules_nodejs/pull/2276/commits/fc78dd68b863138b7c2a0169c5267e93f6885456 for a fix

Comment (bshimc) [reply examples/protocol_buffers/BUILD.bazel:28]: Doh. Thanks.  Fixed in PR, the test failures are now more in line with what others have been reporting.

**Comment by bshimc** [comment]
The test failures are now more in line with what others have reported:

```
$ bazel test //...
INFO: Invocation ID: b97d9c77-b216-4f9a-8990-0263833675c6
INFO: Analyzed 22 targets (0 packages loaded, 0 targets configured).
INFO: Found 17 targets and 5 test targets...
ERROR: /Users/bo/src/github.com/bazelbuild/rules_nodejs/examples/protocol_buffers/BUILD.bazel:86:14: Bundling JavaScript bundle.js [rollup] failed (Exit 1): rollup.sh failed: error executing command bazel-out/host/bin/external/npm/rollup/bin/rollup.sh bazel-out/darwin-fastbuild/bin/app.mjs --output.file bazel-out/darwin-fastbuild/bin/bundle.js --format iife --config ... (remaining 6 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox rollup.sh failed: error executing command bazel-out/host/bin/external/npm/rollup/bin/rollup.sh bazel-out/darwin-fastbuild/bin/app.mjs --output.file bazel-out/darwin-fastbuild/bin/bundle.js --format iife --config ... (remaining 6 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox

bazel-out/darwin-fastbuild/bin/app.mjs → bazel-out/darwin-fastbuild/bin/bundle.js...
[!] Error: Could not resolve './google/api/annotations_pb' from bazel-out/darwin-fastbuild/bin/car_grpc_web_pb.mjs
Error: Could not resolve './google/api/annotations_pb' from bazel-out/darwin-fastbuild/bin/car_grpc_web_pb.mjs
    at error (/private/var/tmp/_bazel_bo/27e51bb136515535a0b392627bd874c9/sandbox/darwin-sandbox/20/execroot/examples_protocol_buffers/node_modules/rollup/dist/shared/rollup.js:10152:30)
    at ModuleLoader.handleResolveId (/private/var/tmp/_bazel_bo/27e51bb136515535a0b392627bd874c9/sandbox/darwin-sandbox/20/execroot/examples_protocol_buffers/node_modules/rollup/dist/shared/rollup.js:16821:24)
    at /private/var/tmp/_bazel_bo/27e51bb136515535a0b392627bd874c9/sandbox/darwin-sandbox/20/execroot/examples_protocol_buffers/node_modules/rollup/dist/shared/rollup.js:16709:26
    at async Promise.all (index 1)
    at async Promise.all (index 0)
    at async Promise.all (index 0)
    at async Promise.all (index 0)
    at async Promise.all (index 0)
    at rollupInternal (/private/var/tmp/_bazel_bo/27e51bb136515535a0b392627bd874c9/sandbox/darwin-sandbox/20/execroot/examples_protocol_buffers/node_modules/rollup/dist/shared/rollup.js:18221:18)
    at build (/private/var/tmp/_bazel_bo/27e51bb136515535a0b392627bd874c9/sandbox/darwin-sandbox/20/execroot/examples_protocol_buffers/node_modules/rollup/dist/bin/rollup:1440:20)

INFO: Elapsed time: 3.950s, Critical Path: 3.59s
INFO: 5 processes: 5 internal.
FAILED: Build did NOT complete successfully
//:dummy_test                                                   (cached) PASSED in 0.3s
//:devserver_test_chromium-local                                      NO STATUS
//:prodserver_test_chromium-local                                     NO STATUS
//:test_chromium-local                                                NO STATUS
//:test_firefox-local                                                 NO STATUS

Executed 0 out of 5 tests: 1 test passes and 4 were skipped.
FAILED: Build did NOT complete successfully
```

**Comment by mrmeku** [comment]
@bshimc Thanks for the update, I'll look into the root cause soon

**Comment by gonzojive** [comment]
It would be even better if the updated test case(s) involved .ts files that import an external ts protobuf using a non-annotation proto type.

The case I'm interested in is when you'd like to import a proto library for use in a `.ts` file, but there is no explicit ts_proto_library rule for the library. Right now when I try to do this, I get an error like

```
transitive dependency on bazel-out/k8-fastbuild/bin/external/go_googleapis/google/type/latlng_pb.d.ts not allowed. Please add the BUILD target to your rule's deps.
```

**Comment by bshi** [comment]
reopen

**Pull request closed without merging**